crates/primitives/starknet/src/transaction/execution.rs

```diff
@@ -0,0 +1,566 @@
+
+        // execute
+        let ValidateExecuteCallInfo { validate_call_info, execute_call_info, revert_error } = self.execute_inner(
+            state,
+            block_context,
+            &mut execution_resources,
+            &mut remaining_gas,
+            &account_tx_context,
+        )?;
+
+        let (actual_fee, fee_transfer_call_info, actual_resources) = self.handle_fee(
+            state,
+            &execute_call_info,
+            &validate_call_info,
+            &mut execution_resources,
+            block_context,
+            account_tx_context,
+        )?;
+
+        let tx_execution_info = TransactionExecutionInfo {
+            validate_call_info,
+            execute_call_info,
+            fee_transfer_call_info,
+            actual_fee,
+            actual_resources,
+            revert_error,
+        };
+
+        Ok(tx_execution_info)
+    }
+
+    fn handle_fee<S: State + StateChanges + FeeConfig>(
+        &self,
+        state: &mut S,
+        execute_call_info: &Option<CallInfo>,
+        validate_call_info: &Option<CallInfo>,
+        execution_resources: &mut ExecutionResources,
+        block_context: &BlockContext,
+        account_tx_context: AccountTransactionContext,
+    ) -> TransactionExecutionResult<(Fee, Option<CallInfo>, ResourcesMapping)> {
+        let actual_resources = compute_transaction_resources(
+            state,
+            execute_call_info,
+            validate_call_info,
+            execution_resources,
+            Self::tx_type(),
+            None,
+        )?;
+
+        let (actual_fee, fee_transfer_call_info) =
+            charge_fee(state, block_context, account_tx_context, &actual_resources)?;
+
+        Ok((actual_fee, fee_transfer_call_info, actual_resources))
+    }
+}
+
+impl Validate for InvokeTransaction {
+    const VALIDATE_TX_ENTRY_POINT_NAME: &'static str = VALIDATE_ENTRY_POINT_NAME;
+}
+
+impl Execute for InvokeTransaction {
+    fn execute_inner<S: State + StateChanges + FeeConfig>(
+        &self,
+        state: &mut S,
+        block_context: &BlockContext,
+        resources: &mut ExecutionResources,
+        remaining_gas: &mut u64,
+        account_tx_context: &AccountTransactionContext,
+    ) -> TransactionExecutionResult<ValidateExecuteCallInfo> {
+        let mut context = EntryPointExecutionContext::new(
+            block_context.clone(),
+            account_tx_context.clone(),
+            block_context.invoke_tx_max_n_steps as usize,
+        );
+
+        let validate_call_info = self.validate_tx_inner(
+            state,
+            resources,
+            remaining_gas,
+            &mut context,
+            GetTransactionCalldata::calldata(self),
+        )?;
+        let validate_execute_call_info = match self.tx {
+            // V0 tx cannot revert, we cannot charge the failling ones
+            starknet_api::transaction::InvokeTransaction::V0(_) => {
```

Comment:
It's a good question.
I think we should remove it.
But what about projects like deoxys. They need to keep executing those old tx, don't they?

Reply:
I send them a message. But we may need to keep supporting v0 tx for them. Maybe behind a feature flag to not bother other usages

Reply:
Starknet mainnet doesn't have V0 as well as mentioned [here](https://docs.starknet.io/documentation/starknet_versions/deprecated/#removed_features). I don't think V0 has anything to do with interacting with the old state so why would deoxys need it?

Reply:
Deoxys is reexecuting all tx in blocks in order to sync state.
So they have to execute invokev0 tx in order to process old blocks. It may be problem for us as we want to update ours deps to the later versions but those droped invokev0 support

Reply:
But why would deoxys not take the latest proved state from the L1. It doesn't need to execute all the transaction in that case.

In either case, maybe we can put it behind a feature flag then so that Invoke0 is disabled by default?

Reply:
Just to confirm, we aren't supporting V0 for now right? And if not, should we throw an error here?

Reply:
We support executing them internally (blockifier) but not receiving them through RPC.
Meaning somebody can still execute them by calling the extrinsic directly, something we should disable.

But we want to keep being able to execute them for Full node sync purpose

Reply:
Oh ok, got it. Cool.